tests/ui/README.md

```diff
@@ -0,0 +1,123 @@
+- Components that change their state or appearance based on props or state require tests to ensure they adapt correctly.
+- Example: A dropdown that expands when clicked.
+
+``` javascript
+test('expands dropdown when clicked', () => {
+    render(<Dropdown label="Options" options={['Option 1', 'Option 2']} />);
+    expect(screen.queryByText('Option 1')).not.toBeInTheDocument();
+
+    fireEvent.press(screen.getByText('Options'));
+    expect(screen.getByText('Option 1')).toBeInTheDocument();
+    expect(screen.getByText('Option 2')).toBeInTheDocument();
+});
+```
+
+3. **Edge Cases**:
+- It's crucial to test how your component behaves with invalid or unusual inputs to ensure stability and handle user errors gracefully
+- Example: Testing an input field's behavior with empty or invalid values.
+
+``` javascript
+test('shows error message for invalid input', () => {
+    render(<TextInputWithValidation />);
+
+    const input = screen.getByPlaceholderText('Enter your email');
+    fireEvent.changeText(input, 'invalid-email');
+
+    expect(screen.getByText('Please enter a valid email')).toBeInTheDocument();
+});
+```
+
+4. **Reusable UI Patterns**:
+- Why: Components that are reused across the app need consistent behavior across different contexts.
+- Example: Custom Checkbox or Dropdown components.
+
+``` javascript
+test('toggles state when clicked', () => {
+    const mockOnChange = jest.fn();
+    render(<Checkbox label="Accept Terms" onChange={mockOnChange} />);
+
+    const checkbox = screen.getByText('Accept Terms');
+    fireEvent.press(checkbox);
+    expect(mockOnChange).toHaveBeenCalledWith(true);
+
+    fireEvent.press(checkbox);
+    expect(mockOnChange).toHaveBeenCalledWith(false);
+});
+```
+### When to Skip UI Tests for Components:
+
+**Purely Presentational Components**:
+
+- Why: These components don’t contain logic or interactivity, so testing them is often unnecessary. Focus on visual output and accessibility instead.
+- Example: Avatar component that only displays an image, we generally don’t need tests unless there’s a specific behavior to verify.
+
+### Do’s
+
+- Write tests that reflect actual user behavior (e.g., clicking buttons, filling forms).
+- Mock external dependencies like network calls or Onyx data.
+- Use `findBy` and `waitFor` to handle asynchronous updates in the UI.
+- Follow naming conventions for test IDs (`testID="component-name-action"`).
+- Test for accessibility attributes like `accessibilityLabel`, `accessibilityHint`, etc., to improve test coverage for screen readers.
+- Reuse test utilities: Write helper functions to handle repetitive test setup, reducing code duplication and improving maintainability.
+- **Rather than targeting 100% coverage, prioritize meaningful tests for critical workflows and components**.
+
+### Don’ts
+
+- Don’t test implementation details (e.g., state variables or internal method calls).
+- Don’t ignore warnings: Fix `act()` or other common warnings to ensure reliability.
```

Comment:
Could you please add more guidance for `act()`? It has always confused me

Reply:
Added a short explanation and clarified when manual usage is needed